### PR TITLE
Swap $key and $index parameters in ChoiceType

### PR DESCRIPTION
**Pull request by technetium**
I've noticed the second parameter in the callback function for the
options in the ChoiceType represents the index and the third the
key.

This patch brings the documentation in line with the code.

### CODE REVIEW

File: reference/forms/types/choice.rst [hunk 1]
```diff
@@ -92,19 +92,18 @@ method::
             new Category('Cat3'),
             new Category('Cat4'),
         ],
-        'choice_label' => function($category, $key, $index) {
+        'choice_label' => function($category, $index, $key) {
```
Comment (xabbuh) [inline reference/forms/types/choice.rst]: `function($category, $key, $value)`

File: reference/forms/types/choice.rst [hunk 2]
```diff
@@ -92,19 +92,18 @@ method::
...
             /** @var Category $category */
             return strtoupper($category->getName());
         },
-        'choice_attr' => function($category, $key, $index) {
+        'choice_attr' => function($category, $index, $key) {
```
Comment (xabbuh) [inline reference/forms/types/choice.rst]: `function($category, $key, $value)`

File: reference/forms/types/choice.rst [hunk 3]
```diff
@@ -92,19 +92,18 @@ method::
...
             return ['class' => 'category_'.strtolower($category->getName())];
         },
-        
-        'group_by' => function($category, $key, $index) {
+        'group_by' => function($category, $index, $key) {
```
Comment (xabbuh) [inline reference/forms/types/choice.rst]: `function($category, $key, $value)`

File: reference/forms/types/options/choice_attr.rst.inc [hunk 4]
```diff
@@ -18,7 +18,7 @@ If an array, the keys of the ``choices`` array must be used as keys::
             'No' => false,
             'Maybe' => null,
         ),
-        'choice_attr' => function($val, $key, $index) {
+        'choice_attr' => function($val, $index, $key) {
```
Comment (xabbuh) [inline reference/forms/types/options/choice_attr.rst.inc]: `function($category, $key, $value)`

File: reference/forms/types/options/choice_label.rst.inc [hunk 5]
```diff
@@ -16,7 +16,7 @@ more control::
             'no' => false,
             'maybe' => null,
         ),
-        'choice_label' => function ($value, $key, $index) {
+        'choice_label' => function ($value, $index, $key) {
```
Comment (xabbuh) [inline reference/forms/types/options/choice_label.rst.inc]: `function($category, $key, $value)`

File: reference/forms/types/options/group_by.rst.inc [hunk 6]
```diff
@@ -22,7 +22,7 @@ Take the following example::
             '1 week' => new \DateTime('+1 week'),
             '1 month' => new \DateTime('+1 month')
         ),
-        'group_by' => function($val, $key, $index) {
+        'group_by' => function($val, $index, $key) {
```
Comment (xabbuh) [inline reference/forms/types/options/group_by.rst.inc]: `function($category, $key, $value)`

**Review by vudaltsov — CHANGES_REQUESTED** [review]
Agree with @xabbuh, 2 and 3 arguments should be named `$key` and `$value`, the first argument should be named according to choice data.

**Comment by javiereguiluz** [comment]
I'm afraid this PR stalled and now it has some conflicts. So I've opened #9835 to finish it. Thanks @technetium for opening this and thanks to reviewers.

**Pull request closed without merging**